src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -3052,27 +3052,191 @@ partial class C
+                //     public partial string M2() => "hello";
+                Diagnostic(ErrorCode.ERR_PartialMethodSignatureDifference, "M2").WithArguments("string? C.M2()", "string C.M2()").WithLocation(9, 27),
                 // (13,26): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
                 //     public partial string? M3() => null; // 2
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(13, 26),
+                // (13,28): error CS8824: Partial method declarations 'string C.M3()' and 'string? C.M3()' must have identical parameter types and identical return types.
+                //     public partial string? M3() => null; // 2
+                Diagnostic(ErrorCode.ERR_PartialMethodSignatureDifference, "M3").WithArguments("string C.M3()", "string? C.M3()").WithLocation(13, 28),
                 // (15,26): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
                 //     public partial string? M4(); // 3
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(15, 26),
+                // (16,27): error CS8824: Partial method declarations 'string? C.M4()' and 'string C.M4()' must have identical parameter types and identical return types.
+                //     public partial string M4() => "hello";
+                Diagnostic(ErrorCode.ERR_PartialMethodSignatureDifference, "M4").WithArguments("string? C.M4()", "string C.M4()").WithLocation(16, 27),
+                // (20,27): error CS8824: Partial method declarations 'string C.M6()' and 'string C.M6()' must have identical parameter types and identical return types.
+                //     public partial string M6() => null!;
+                Diagnostic(ErrorCode.ERR_PartialMethodSignatureDifference, "M6").WithArguments("string C.M6()", "string C.M6()").WithLocation(20, 27),
+                // (22,27): error CS8824: Partial method declarations 'string? C.M8()' and 'string C.M8()' must have identical parameter types and identical return types.
+                //     public partial string M8() => null!;
+                Diagnostic(ErrorCode.ERR_PartialMethodSignatureDifference, "M8").WithArguments("string? C.M8()", "string C.M8()").WithLocation(22, 27),
+                // (24,28): error CS8824: Partial method declarations 'string C.M10()' and 'string? C.M10()' must have identical parameter types and identical return types.
+                //     public partial string? M10() => null;
+                Diagnostic(ErrorCode.ERR_PartialMethodSignatureDifference, "M10").WithArguments("string C.M10()", "string? C.M10()").WithLocation(24, 28),
+                // (29,27): error CS8824: Partial method declarations 'string C.M5()' and 'string C.M5()' must have identical parameter types and identical return types.
```

Comment:
As a user I would be very confused if I saw this diagnostic. Since many code generators have nullability disabled, this may be a scenario that matters. It's not clear to me if code generators which adopt extended partial methods are also expected to adopt nullability in sync with their users.

I think we may want oblivious nullability to compare "equal" to annotated and to not-annotated nullability for purposes of partial method checks. One thing that may lack tests here currently is when the declaration part is oblivious and the implementation part is nullable-enabled.

Reply:
Synced offline, currently I'm alright with the "oblivious generated signature" and "nullable-enabled user implementation" scenario generating a warning that requires surrounding the signature with `#nullable disable annotations`/`#nullable enable annotations`. We can adjust if we learn there is an undesirable impact from doing so.

Reply:
We decided in LDM to keep most warnings/errors the way they are, and to add a warning wave warning to "any difference between partial methods, except for nullable/oblivious differences". https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-09-14.md#partial-method-signature-matching